docs(NotificationBadge): add accessibility documentation (#4766)
kiwicom/orbit: docs/src/documentation/03-components/03-information/notificationbadge/03-accessibility.mdx

The following props are available to improve the accessibility of your NotificationBadge component:

| Name      | Type         | Description                                                                                          |
| :-------- | :----------- | :--------------------------------------------------------------------------------------------------- |
| ariaLabel | `string`     | Provides an accessible name for the notification badge that will be announced by screen readers.     |
| icon      | `React.Node` | The displayed icon. When used, `ariaLabel` should be provided to ensure screen reader accessibility. |

### Automatic Accessibility Features

The NotificationBadge component automatically sets `role="status"` to announce updates to assistive technologies.

### The `ariaLabel` Prop

The `ariaLabel` prop is crucial for making NotificationBadge components accessible:

- It sets the `aria-label` attribute on the badge, providing context for screen readers
- When provided, it completely replaces what screen readers would announce about the badge's visual content
- It's essential when using icons or when the numerical content alone isn't sufficient to convey meaning

> - For number-only badges (like "3"), it adds context about what the number represents (e.g., "3 unread messages")

Reviewer: ```suggestion
- For number-only badges (like "3"), it can be used to add context about what the number represents (e.g., "3 unread messages")
```

### Best Practices

#### Screen Reader Support

> - Always provide a descriptive `ariaLabel` that clearly explains what the notification represents

Reviewer: ```suggestion
- Always provide a descriptive `ariaLabel` that clearly explains what the notification badge represents
```

- Ensure all `ariaLabel` text is translated to the user's language
- Make the label concise but informative, focusing on the count and its meaning

#### Using Icons

> - When using the `icon` prop, always provide an `ariaLabel` since the icon will replace any children content

Reviewer: I had the same question when reading the table above: should I use `ariaLabel` on the icon or on the `NotificationBadge`?

Author: 1. If you set `ariaLabel` on the icon, it will be read as "<ariaLabel text>, image"
2. If you set `ariaLabel` on the `NotificationBadge`, it will be read as "<ariaLabel text>, status" and then "image"

So I would say the best practice would be to use it on the `NotificationBadge` and set `ariaHidden` on the icon, I think I already have such example, will add it to the best practices section.

- Choose icons that visually reinforce the meaning of the notification

> - Remember that icon-only badges without `ariaLabel` will be effectively invisible to screen reader users

Reviewer: Will they? 🤔

Author: Well 😅 Not invisible as the word "image" will be said... But yeah, I will remove this sentence, I think other info is enough.

#### Visual Design Considerations

- Do not rely on color alone to convey the meaning of notifications
- While color types (critical, warning, info, etc.) help visually distinguish importance, always ensure meaning is conveyed through additional means like text, icons, or aria-label

> - This approach ensures users with color vision deficiencies can understand the notification's importance

Reviewer: ```suggestion
- This approach ensures users with color vision impairments can understand the notification's importance
```

Slightly nicer
